src/client/dfs/duns.c

```diff
@@ -846,47 +914,75 @@ duns_create_path(daos_handle_t poh, const char *path, struct duns_attr_t *attrp)
 	}
 
 	if (attrp->da_type == DAOS_PROP_CO_LAYOUT_POSIX) {
-		struct statfs   fs;
-		char            *dir, *dirp;
-		mode_t		mode = S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH;
+		struct statfs fs;
+		char         *dir, *dirp;
+		mode_t        mode = S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH;
+
+#ifdef LUSTRE_INCLUDE
+		if (fs.f_type == LL_SUPER_MAGIC) {
+			rc = duns_create_lustre_path(poh, info, path, mode, attrp);
+			if (rc == 0)
+				return 0;
+			/* if Lustre specific method fails, fallback to try the normal way... */
+		}
+#endif
 
 		D_STRNDUP(dir, path, path_len);
 		if (dir == NULL) {
-			D_ERROR("Failed copy path %s: %s\n", path, strerror(errno));
```

Comment:
remove braces then

Reply:
Done.  I think we have both but I do prefer without braces myself.